### PR TITLE
Fix cppcheck warnings

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR fixes some of the remaining cppcheck suppressions

**To test:**
Code review and make sure the tests pass.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/General/StepScan.cpp [hunk 1]
```diff
@@ -599,6 +599,7 @@ void StepScan::generateCurve(const QString &var) {
     MatrixWorkspace_sptr top = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(m_plotWSName);
     MatrixWorkspace_sptr bottom = norm->getProperty("OutputWorkspace");
     top /= bottom;
+    AnalysisDataService::Instance().addOrReplace(m_plotWSName, top);
```
Comment (DanielMurphy22) [inline qt/scientific_interfaces/General/StepScan.cpp:602]: I've thought about whether adding an explicit call to divide is worth it or not. Sadly the OutputWorkspace name from norm (`bottom->getName()`) returns an empty string and the bottom workspace is not in the ADS. So I think I'm happy with this addOrReplace call as it's in a relatively niche part of the codebase.

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
Really good job! It's important to get these bits of the codebase tidied up and it prevents serious bugs appearing in the future. Just the small change!

**Review by DanielMurphy22 — DISMISSED** [review]

**Comment by robertapplin** [comment]
@mantidproject/gatekeepers the test failure on this is unrelated and should be fixed in #34162

File: qt/scientific_interfaces/General/StepScan.cpp (hunk 1 above)
Comment (martyngigg) [reply qt/scientific_interfaces/General/StepScan.cpp:602]: Can I ask how this came about as a fix for a cppcheck warning?

**Review by martyngigg — COMMENTED** [review]
I'm happy with everything with the exception of one thing that wasn't immediately obvious to me.

File: qt/scientific_interfaces/General/StepScan.cpp (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/General/StepScan.cpp:602]: Cppcheck was complaining that top is an `unreadVariable`. I did something a bit cheaty where I add the top workspace into the ADS (even though it already exists in the ADS), just to remove the unreadVariable warning. I'm happy to remove this and put back in the cppcheck suppression as its probably a false positive

**Review by martyngigg — CHANGES_REQUESTED** [review]
Just marking this as a request for changes now.

**Review by DanielMurphy22 — DISMISSED** [review]
Sensible to not change functionality!

**Review by DanielMurphy22 — APPROVED** [review]

**Review by martyngigg — APPROVED** [review]

**Pull request merged**